### DISCUSSION TITLE
Commit action in Cisco IOS XR

### DISCUSSION THREAD

**Discussion by jonascollen** [scrapli/scrapligo#182] [Q&A] [answered]
I'm trying to figure out if there is any built in function for commiting configuration in IOS XR?

Currently, sending config with d.SendConfigs(cmds) will just "hang" if I also dont include a 'commit' in the end of my commands.
It would be nice if there was a function to use d.Commit or similar so I can catch errors related to the commit-action only, but maybe there is and I just haven't found it?

For reference;

```Go
	p, err := platform.NewPlatform(
		"cisco_iosxr",
		host,
		options.WithAuthNoStrictKey(),
		options.WithAuthUsername(settings.Username),
		options.WithAuthPassword(settings.Password),
	)

	if err != nil {
		return err
	}

	d, err := p.GetNetworkDriver()
	if err != nil {
		return err
	}

	err = d.Open()
	if err != nil {
		return err
	}

	defer d.Close()
	
       	cmds := []string{"interface Loopback0", "description Configured by Go", "commit"}
       	
	_, err = d.SendConfigs(cmds)
	if err != nil {
		return err
	}

	return nil
```

**Comment by carlmontanari** [comment, marked as answer]
nope, you're not missing anything. scrapli(go) doesn't care about this basically -- its up to you. scrapli(go)cfg does this for you though. 

if you want to just have the commit things handled by itself so you can see errors related to that only, you could just send another sendconfigs after you send all the config lines. scrapli wont leave config mode till you ask it to do a "command" or otherwise acquire a different privilege level.

**Comment by jonascollen** [comment]
I see, thank you for the clarification.
